### PR TITLE
mes-1774: Adding view components for new email and email provided at the time o…

### PR DESCRIPTION
**Pull request by MarkPatt89**
…f booking on the

communication page.

 Reducer, actions and selector for communication preferences.(How the candidate wishes
to receive their results).

## Description and relevant Jira numbers

- Created two components. One for emails provided at the time of booking, the other to input a new email address.
- Implemented a reducer, actions and selector for the ```CommunicationPreferences``` entity in the schema.
- Dispatch an action when a candidate selects the provided email.
- Dispatch an action when the candidate inputs a new email. 
- Validation on new email input to ensure it is a valid email format
- Added the new reducer to ```combineReducers()```

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [x] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

Screenshot 2019-05-16 at 08 25 44

Screenshot 2019-05-16 at 08 26 07

Screenshot 2019-05-16 at 08 26 19

### CODE REVIEW

File: src/modules/tests/communication-preferences/communication-preferences.reducer.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  communicationMethod: null,
+};
+
+export const communicationPreferencesReducer = (
+  state = initialState,
+  action: communicationPrefActions.Types,
+): CommunicationPreferences => {
+  switch (action.type) {
+    case communicationPrefActions.CANDIDATE_CONFIRMED_COMMUNICATION_PREFERENCE_AS_EMAIL:
+      return {
```
Comment (roberthall12) [inline src/modules/tests/communication-preferences/communication-preferences.reducer.ts]: do you need to return `...state` here?

File: src/pages/communication-page/communication.ts [hunk 2]
```diff
@@ -18,26 +18,55 @@ import {
...
-  form: FormGroup;
+  readonly providedEmail = 'providedEmail';
+  readonly newEmail = 'newEmail';
 
+  @ViewChild(Navbar)
+  navBar: Navbar;
+
+  form: FormGroup;
+  subscription: Subscription;
+  communicationChoice: string;
```
Comment (roberthall12) [inline src/pages/communication-page/communication.ts]: can this be an enum?

File: src/modules/tests/communication-preferences/communication-preferences.reducer.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        ...state,
+        communicationMethod: action.communicationMethod,
+      };
+    case communicationPrefActions.CANDIDATE_CONFIRMED_COMMUNICATION_PREFERENCE_AS_SUPPORT_CENTRE:
+      return {
+        ...state,
+        communicationMethod: action.communicationMethod,
+      };
+  }
+  return state;
```
Comment (Cavalier07) [inline src/modules/tests/communication-preferences/communication-preferences.reducer.ts]: this should be in the default branch of the switch.

File: src/pages/communication-page/communication.scss [hunk 4]
```diff
@@ -2,4 +2,42 @@ communication {
   ion-content>div {
     background-color: map-get($colors, "mes-white");
   }
+
+  .communication-option {
+    padding: 10px 5px 10px 5px;
```
Comment (Cavalier07) [inline src/pages/communication-page/communication.scss]: can be shorthanded

**Comment by Cavalier07** [comment]
Looks good - need to get the schema updated in npm else this will fail the build

**Pull request merged**